scripts/Diffraction/isis_powder/routines/focus.py

```diff
@@ -48,14 +51,43 @@ def _focus_one_ws(input_workspace, run_number, instrument, perform_vanadium_norm
         summed_empty = common.generate_summed_runs(empty_sample_ws_string=run_details.sample_empty,
                                                    instrument=instrument,
                                                    scale_factor=instrument._inst_settings.sample_empty_scale)
-    if summed_empty is not None:
+
+    # Correct for absorption / multiple scattering if required
+    if paalman_pings and summed_empty and absorb and container_details:
+        events_per_point_string = instrument._inst_settings.paalman_pings_events_per_point
+        if events_per_point_string:
+            events_per_point = int(events_per_point_string)
+        else:
+            events_per_point = 1000
+        expected_filename = "PaalmanPingsApplied_" + instrument.get_instrument_prefix() + run_number \
+                            + "_" + instrument._get_input_batching_mode() \
+                            + f"_{events_per_point}eventsperpoint" + ".nxs"
+        save_directory = instrument._inst_settings.output_dir
+        if expected_filename in os.listdir(save_directory):
+            input_workspace = mantid.LoadNexus(os.path.join(save_directory, expected_filename))
+        else:
+            corrections = mantid.PaalmanPingsMonteCarloAbsorption(
+                InputWorkspace=input_workspace,
+                Shape=sample_details._shape_type.capitalize(),
```

Comment:
I think we should discuss this point - but I'll pose the question here at least:
In other parts of the code that sample_details class is used to prepare the inputs to a call to the SetSample algorithm. An absorption correction algorithm is then called soon after that.
I wonder if it would make things more consistent if you called SetSample here and then used the Shape=Preset option in the PaalmanPingsMonteCarloAbsorption algorithm